src/battle_ai_main.c

```diff
@@ -1762,6 +1762,7 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
             break;
         case EFFECT_FOLLOW_ME:
         case EFFECT_HELPING_HAND:
+        case EFFECT_DRAGON_CHEER:
             if (!isDoubleBattle
               || !IsBattlerAlive(BATTLE_PARTNER(battlerAtk))
               || PartnerHasSameMoveEffectWithoutTarget(BATTLE_PARTNER(battlerAtk), move, aiData->partnerMove)
```

Comment:
The conditions from here down are all irrelevant to Dragon Cheer. Both Pokemon can use Dragon Cheer on each other, the partner's critical hit rate can be boosted for a future turn where they may use a damaging move, and even if the partner switches out, the new partner will still benefit.